diff --git a/src/lib/rulesdata/ancestries/traits.ts b/src/lib/rulesdata/ancestries/traits.ts
index 29c5651..29c5d75 100644
--- a/src/lib/rulesdata/ancestries/traits.ts
+++ b/src/lib/rulesdata/ancestries/traits.ts
@@ -23,15 +23,9 @@ export const traitsData: Trait[] = [
 		cost: 2,
 		effects: [
 			{
-				type: 'MODIFY_STAT',
-				target: 'skillPoints',
+				type: 'INCREASE_SKILL_MASTERY_CAP',
+				count: 1,
 				value: 1
-			},
-			{
-				type: 'GRANT_SKILL_EXPERTISE',
-				target: 'any_skill',
-				value: { capIncrease: 1, levelIncrease: 1 },
-				userChoice: { prompt: 'Choose a skill for Expertise' }
 			}
 		]
 	},
@@ -57,10 +51,9 @@ export const traitsData: Trait[] = [
 		cost: 1,
 		effects: [
 			{
-				type: 'GRANT_TRADE_EXPERTISE',
-				target: 'any_trade',
-				value: { capIncrease: 1, levelIncrease: 1 },
-				userChoice: { prompt: 'Choose a Trade for Expertise' }
+				type: 'INCREASE_TRADE_MASTERY_CAP',
+				count: 1,
+				value: 1
 			}
 		]
 	},
@@ -477,10 +470,9 @@ export const traitsData: Trait[] = [
 		cost: 1,
 		effects: [
 			{
-				type: 'GRANT_TRADE_EXPERTISE',
-				target: 'any_trade',
-				value: { capIncrease: 1, levelIncrease: 1 },
-				userChoice: { prompt: 'Choose a Trade for Expertise' }
+				type: 'INCREASE_TRADE_MASTERY_CAP',
+				count: 1,
+				value: 1
 			}
 		]
 	},
@@ -640,10 +632,33 @@ export const traitsData: Trait[] = [
 		cost: 1,
 		effects: [
 			{
-				type: 'GRANT_TRADE_EXPERTISE',
-				target: 'crafting_or_subterfuge_trade',
-				value: { capIncrease: 1, levelIncrease: 1 },
-				userChoice: { prompt: 'Choose a Crafting or Subterfuge Trade for Expertise' }
+				type: 'INCREASE_TRADE_MASTERY_CAP',
+				count: 1,
+				value: 1,
+				options: [
+					'alchemy',
+					'blacksmithing',
+					'brewing',
+					'calligraphy',
+					'carpentry',
+					'cartography',
+					'cobbling',
+					'cooking',
+					'disguise',
+					'forgery',
+					'herbalism',
+					'jeweler',
+					'leatherworking',
+					'masonry',
+					'painting',
+					'poisoner',
+					'pottery',
+					'sculpting',
+					'smithing',
+					'tailoring',
+					'thieves',
+					'woodcarving'
+				]
 			}
 		]
 	},
@@ -867,16 +882,16 @@ export const traitsData: Trait[] = [
 		]
 	},
 	{
-		id: 'beastborn_natural_weapons',
-		name: 'Natural Weapons',
+		id: 'beastborn_natural_weapon',
+		name: 'Natural Weapon',
 		description:
-			'Your claws and teeth are Natural Weapons that deal 1 Slashing or Piercing damage.',
+			'You have up to 2 Natural Weapons (claws, horns, fangs, tail, etc.) which you can use to make Unarmed Strikes that deal 1 Bludgeoning, Piercing, or Slashing damage (your choice upon gaining this Trait). You can perform Attack Maneuvers with your Natural Weapons.',
 		cost: 1,
 		effects: [
 			{
 				type: 'GRANT_ABILITY',
-				target: 'natural_weapons',
-				value: 'Claws and teeth are Natural Weapons dealing 1 damage.'
+				target: 'natural_weapon',
+				value: '2 Natural Weapons for Unarmed Strikes (1 damage, chosen type).'
 			}
 		]
 	},
@@ -953,7 +968,7 @@ export const traitsData: Trait[] = [
 		name: 'Mana Increase',
 		description: 'Your MP maximum increases by 1.',
 		cost: 1,
-		effects: [{ type: 'MODIFY_STAT', target: 'mp', value: 1 }]
+		effects: [{ type: 'MODIFY_STAT', target: 'mpMax', value: 1 }]
 	},
 	{
 		id: 'angelborn_celestial_clarity',
@@ -1049,10 +1064,33 @@ export const traitsData: Trait[] = [
 		cost: 1,
 		effects: [
 			{
-				type: 'GRANT_TRADE_EXPERTISE',
-				target: 'any_crafting_or_services_trade',
-				value: { capIncrease: 1, levelIncrease: 1 },
-				userChoice: { prompt: 'Choose a Crafting or Services Trade for Expertise' }
+				type: 'INCREASE_TRADE_MASTERY_CAP',
+				count: 1,
+				value: 1,
+				options: [
+					'alchemy',
+					'blacksmithing',
+					'brewing',
+					'calligraphy',
+					'carpentry',
+					'cartography',
+					'cobbling',
+					'cooking',
+					'herbalism',
+					'jeweler',
+					'leatherworking',
+					'masonry',
+					'painting',
+					'pottery',
+					'sculpting',
+					'smithing',
+					'tailoring',
+					'woodcarving',
+					'music',
+					'navigation',
+					'vehicles_land',
+					'vehicles_water'
+				]
 			}
 		]
 	},
@@ -1699,8 +1737,7 @@ export const traitsData: Trait[] = [
 			{
 				type: 'GRANT_ABILITY',
 				target: 'natural_weapon',
-				value: '2 Natural Weapons for Unarmed Strikes (1 damage, chosen type).',
-				userChoice: { prompt: 'Choose damage type: Bludgeoning, Piercing, or Slashing' }
+				value: '2 Natural Weapons for Unarmed Strikes (1 damage, chosen type).'
 			}
 		]
 	},
@@ -1743,8 +1780,7 @@ export const traitsData: Trait[] = [
 			{
 				type: 'GRANT_ADV_ON_CHECK',
 				target: 'Awareness',
-				value: 'chosen_sense',
-				userChoice: { prompt: 'Choose a sense: hearing, sight, or smell' }
+				value: 'chosen_sense'
 			}
 		]
 	},
@@ -1862,8 +1898,7 @@ export const traitsData: Trait[] = [
 			{
 				type: 'GRANT_ABILITY',
 				target: 'hazardous_hide',
-				value: 'Grapplers take 1 damage (chosen type) per turn.',
-				userChoice: { prompt: 'Choose damage type: Corrosion, Piercing, or Poison' }
+				value: 'Grapplers take 1 damage (chosen type) per turn.'
 			}
 		]
 	},
@@ -1950,8 +1985,7 @@ export const traitsData: Trait[] = [
 			{
 				type: 'GRANT_ABILITY',
 				target: 'weapon_style_passive',
-				value: 'Natural Weapon gains chosen Weapon Style passive.',
-				userChoice: { prompt: 'Choose a Weapon Style' }
+				value: 'Natural Weapon gains chosen Weapon Style passive.'
 			}
 		]
 	},
@@ -2195,14 +2229,14 @@ export const traitsData: Trait[] = [
 		description: 'Your Size is considered Small.',
 		cost: -1,
 		isNegative: true,
-		effects: [{ type: 'GRANT_ABILITY', target: 'Small-Sized', value: true }]
+		effects: [{ type: 'GRANT_ABILITY', target: 'Small-Sized', value: 'Your Size is considered Small.' }]
 	},
 	{
 		id: 'gremlin_sneaky',
 		name: 'Sneaky',
 		description: 'You can Hide while only Partially Concealed or behind 1/2 Cover.',
 		cost: 2,
-		effects: [{ type: 'GRANT_ABILITY', target: 'Sneaky', value: true }]
+		effects: [{ type: 'GRANT_ABILITY', target: 'Sneaky', value: 'You can Hide while only Partially Concealed or behind 1/2 Cover.' }]
 	},
 	{
 		id: 'gremlin_thriller',
@@ -2210,7 +2244,7 @@ export const traitsData: Trait[] = [
 		description:
 			'Once per Round, when a hostile creature within 5 Spaces becomes Intimidated, Frightened, or Terrified, you gain ADV on the next Check or Save you make.',
 		cost: 2,
-		effects: [{ type: 'GRANT_ABILITY', target: 'Thriller', value: true }]
+		effects: [{ type: 'GRANT_ABILITY', target: 'Thriller', value: 'Once per Round, when a hostile creature within 5 Spaces becomes Intimidated, Frightened, or Terrified, you gain ADV on the next Check or Save you make.' }]
 	},
 	{
 		id: 'gremlin_surprise',
@@ -2219,7 +2253,7 @@ export const traitsData: Trait[] = [
 			"You have ADV on Checks to impose Fear conditions on creatures you're Hidden from.",
 		cost: 1,
 		effects: [
-			{ type: 'GRANT_ADV_ON_CHECK', target: 'Impose_Fear_Conditions_while_Hidden', value: true }
+			{ type: 'GRANT_ADV_ON_CHECK', target: 'Impose_Fear_Conditions_while_Hidden', value: 'ADV' }
 		]
 	},
 	{
@@ -2228,7 +2262,7 @@ export const traitsData: Trait[] = [
 		description:
 			'You can move through the space of a hostile creature 1 size larger than you as Difficult Terrain.',
 		cost: 1,
-		effects: [{ type: 'GRANT_ABILITY', target: 'Deft Footwork', value: true }]
+		effects: [{ type: 'GRANT_ABILITY', target: 'Deft Footwork', value: 'You can move through the space of a hostile creature 1 size larger than you as Difficult Terrain.' }]
 	},
 	{
 		id: 'gremlin_halfling_disguise',
@@ -2236,7 +2270,7 @@ export const traitsData: Trait[] = [
 		description:
 			'You can spend 1 AP to change appearance to look like a Halfling (reverts under stress).',
 		cost: 0,
-		effects: [{ type: 'GRANT_ABILITY', target: 'Halfling Disguise', value: true }]
+		effects: [{ type: 'GRANT_ABILITY', target: 'Halfling Disguise', value: 'You can spend 1 AP to change appearance to look like a Halfling (reverts under stress).' }]
 	},
 	{
 		id: 'gremlin_bravery',
@@ -2244,9 +2278,9 @@ export const traitsData: Trait[] = [
 		description: 'ADV on Saves against Fear conditions.',
 		cost: 2,
 		effects: [
-			{ type: 'GRANT_ADV_ON_SAVE', target: 'Intimidated', value: true },
-			{ type: 'GRANT_ADV_ON_SAVE', target: 'Frightened', value: true },
-			{ type: 'GRANT_ADV_ON_SAVE', target: 'Terrified', value: true }
+			{ type: 'GRANT_ADV_ON_SAVE', target: 'Intimidated', value: 'ADV' },
+			{ type: 'GRANT_ADV_ON_SAVE', target: 'Frightened', value: 'ADV' },
+			{ type: 'GRANT_ADV_ON_SAVE', target: 'Terrified', value: 'ADV' }
 		]
 	},
 	{
@@ -2275,14 +2309,14 @@ export const traitsData: Trait[] = [
 		name: 'Fearsome',
 		description: 'Enhanced Intimidate Action.',
 		cost: 2,
-		effects: [{ type: 'GRANT_ABILITY', target: 'Fearsome', value: true }]
+		effects: [{ type: 'GRANT_ABILITY', target: 'Fearsome', value: 'Enhanced Intimidate Action.' }]
 	},
 	{
 		id: 'gremlin_natural_weapon',
 		name: 'Natural Weapon',
 		description: 'Natural Weapons for Unarmed Strikes.',
 		cost: 1,
-		effects: [{ type: 'GRANT_ABILITY', target: 'Natural Weapon', value: true }]
+		effects: [{ type: 'GRANT_ABILITY', target: 'Natural Weapon', value: 'Natural Weapons for Unarmed Strikes.' }]
 	},
 	{
 		id: 'gremlin_climb_speed',
@@ -2300,7 +2334,7 @@ export const traitsData: Trait[] = [
 			{
 				type: 'GRANT_ADV_ON_CHECK',
 				target: 'Nature_Survival_Animal_involving_Small_Creatures',
-				value: true
+				value: 'ADV'
 			}
 		]
 	},
@@ -2311,7 +2345,7 @@ export const traitsData: Trait[] = [
 		description: 'Your Size is considered Small.',
 		cost: -1,
 		isNegative: true,
-		effects: [{ type: 'GRANT_ABILITY', target: 'Small-Sized', value: true }]
+		effects: [{ type: 'GRANT_ABILITY', target: 'Small-Sized', value: 'Your Size is considered Small.' }]
 	},
 	{
 		id: 'goblin_escape_artist',
@@ -2320,8 +2354,8 @@ export const traitsData: Trait[] = [
 			'You have ADV on Checks and Saves to avoid or escape being Grappled or Restrained.',
 		cost: 2,
 		effects: [
-			{ type: 'GRANT_ADV_ON_CHECK', target: 'Avoid_or_Escape_Grapple_Restrained', value: true },
-			{ type: 'GRANT_ADV_ON_SAVE', target: 'Avoid_or_Escape_Grapple_Restrained', value: true }
+			{ type: 'GRANT_ADV_ON_CHECK', target: 'Avoid_or_Escape_Grapple_Restrained', value: 'ADV' },
+			{ type: 'GRANT_ADV_ON_SAVE', target: 'Avoid_or_Escape_Grapple_Restrained', value: 'ADV' }
 		]
 	},
 	{
@@ -2330,28 +2364,28 @@ export const traitsData: Trait[] = [
 		description:
 			'Once per Round, you can gain ADV on a Melee Attack. You are Exposed on the next Attack against you until the start of your next turn.',
 		cost: 2,
-		effects: [{ type: 'GRANT_ABILITY', target: 'Aggressive', value: true }]
+		effects: [{ type: 'GRANT_ABILITY', target: 'Aggressive', value: 'Once per Round, you can gain ADV on a Melee Attack. You are Exposed on the next Attack against you until the start of your next turn.' }]
 	},
 	{
 		id: 'goblin_eager_for_combat',
 		name: 'Eager for Combat',
 		description: 'You have ADV on Initiative Checks.',
 		cost: 1,
-		effects: [{ type: 'GRANT_ADV_ON_CHECK', target: 'Initiative', value: true }]
+		effects: [{ type: 'GRANT_ADV_ON_CHECK', target: 'Initiative', value: 'ADV' }]
 	},
 	{
 		id: 'goblin_small_fury',
 		name: 'Small Fury',
 		description: 'You deal +1 damage to Large or larger creatures on a Heavy or Critical Hit.',
 		cost: 1,
-		effects: [{ type: 'GRANT_ABILITY', target: 'Small Fury', value: true }]
+		effects: [{ type: 'GRANT_ABILITY', target: 'Small Fury', value: 'You deal +1 damage to Large or larger creatures on a Heavy or Critical Hit.' }]
 	},
 	{
 		id: 'goblin_natural_trapper',
 		name: 'Natural Trapper',
 		description: 'You have ADV on Checks to gather materials for Traps and to build them.',
 		cost: 0,
-		effects: [{ type: 'GRANT_ADV_ON_CHECK', target: 'Gather_and_Build_Traps', value: true }]
+		effects: [{ type: 'GRANT_ADV_ON_CHECK', target: 'Gather_and_Build_Traps', value: 'ADV' }]
 	},
 	{
 		id: 'goblin_hit_and_run',
@@ -2359,14 +2393,14 @@ export const traitsData: Trait[] = [
 		description:
 			'When you Hit a creature with a Melee Attack, you can spend 1 AP to immediately move up to your Speed without provoking Opportunity Attacks from the target.',
 		cost: 2,
-		effects: [{ type: 'GRANT_ABILITY', target: 'Hit and Run', value: true }]
+		effects: [{ type: 'GRANT_ABILITY', target: 'Hit and Run', value: 'When you Hit a creature with a Melee Attack, you can spend 1 AP to immediately move up to your Speed without provoking Opportunity Attacks from the target.' }]
 	},
 	{
 		id: 'goblin_pack_tactics',
 		name: 'Pack Tactics',
 		description: 'You gain an additional +2 bonus to Attacks against creatures you are Flanking.',
 		cost: 2,
-		effects: [{ type: 'GRANT_ABILITY', target: 'Pack Tactics', value: true }]
+		effects: [{ type: 'GRANT_ABILITY', target: 'Pack Tactics', value: 'You gain an additional +2 bonus to Attacks against creatures you are Flanking.' }]
 	},
 	{
 		id: 'goblin_trapper',
@@ -2375,8 +2409,8 @@ export const traitsData: Trait[] = [
 			'You have ADV on Investigation Checks to spot Traps and Trickery Checks to Hide Traps.',
 		cost: 1,
 		effects: [
-			{ type: 'GRANT_ADV_ON_CHECK', target: 'Investigation_to_spot_Traps', value: true },
-			{ type: 'GRANT_ADV_ON_CHECK', target: 'Trickery_to_Hide_Traps', value: true }
+			{ type: 'GRANT_ADV_ON_CHECK', target: 'Investigation_to_spot_Traps', value: 'ADV' },
+			{ type: 'GRANT_ADV_ON_CHECK', target: 'Trickery_to_Hide_Traps', value: 'ADV' }
 		]
 	},
 	{
@@ -2387,10 +2421,33 @@ export const traitsData: Trait[] = [
 		cost: 1,
 		effects: [
 			{
-				type: 'GRANT_TRADE_EXPERTISE',
-				target: 'any',
-				value: { capIncrease: 1, levelIncrease: 1 },
-				userChoice: { prompt: 'Choose a Crafting or Subterfuge Trade for Expertise' }
+				type: 'INCREASE_TRADE_MASTERY_CAP',
+				count: 1,
+				value: 1,
+				options: [
+					'alchemy',
+					'blacksmithing',
+					'brewing',
+					'calligraphy',
+					'carpentry',
+					'cartography',
+					'cobbling',
+					'cooking',
+					'disguise',
+					'forgery',
+					'herbalism',
+					'jeweler',
+					'leatherworking',
+					'masonry',
+					'painting',
+					'poisoner',
+					'pottery',
+					'sculpting',
+					'smithing',
+					'tailoring',
+					'thieves',
+					'woodcarving'
+				]
 			}
 		]
 	},
@@ -2399,14 +2456,14 @@ export const traitsData: Trait[] = [
 		name: 'Mischievous Hands',
 		description: 'You have ADV on Trickery checks made to conceal or steal objects.',
 		cost: 1,
-		effects: [{ type: 'GRANT_ADV_ON_CHECK', target: 'Trickery_Conceal_or_Steal', value: true }]
+		effects: [{ type: 'GRANT_ADV_ON_CHECK', target: 'Trickery_Conceal_or_Steal', value: 'ADV' }]
 	},
 	{
 		id: 'goblin_sneaky',
 		name: 'Sneaky',
 		description: 'You can Hide while only Partially Concealed or behind 1/2 Cover.',
 		cost: 2,
-		effects: [{ type: 'GRANT_ABILITY', target: 'Sneaky', value: true }]
+		effects: [{ type: 'GRANT_ABILITY', target: 'Sneaky', value: 'You can Hide while only Partially Concealed or behind 1/2 Cover.' }]
 	},
 	{
 		id: 'goblin_reckless',
@@ -2431,9 +2488,9 @@ export const traitsData: Trait[] = [
 			'You have ADV on Saves against being Impaired, Petrified, and transformed against your will.',
 		cost: 2,
 		effects: [
-			{ type: 'GRANT_ADV_ON_SAVE', target: 'Impaired', value: true },
-			{ type: 'GRANT_ADV_ON_SAVE', target: 'Petrified', value: true },
-			{ type: 'GRANT_ADV_ON_SAVE', target: 'Involuntary_Transformation', value: true }
+			{ type: 'GRANT_ADV_ON_SAVE', target: 'Impaired', value: 'ADV' },
+			{ type: 'GRANT_ADV_ON_SAVE', target: 'Petrified', value: 'ADV' },
+			{ type: 'GRANT_ADV_ON_SAVE', target: 'Involuntary_Transformation', value: 'ADV' }
 		]
 	},
 	{
@@ -2449,7 +2506,7 @@ export const traitsData: Trait[] = [
 		description:
 			"You can use your hands as Sculptor's Tools or Mason's Tools to shape soil, sand, stone, or rock. When you do, you can use your Prime Modifier instead of the normal Attribute on any Checks you make.",
 		cost: 0,
-		effects: [{ type: 'GRANT_ABILITY', target: 'Mold Earth', value: true }]
+		effects: [{ type: 'GRANT_ABILITY', target: 'Mold Earth', value: 'You can use your hands as Sculptor\'s Tools or Mason\'s Tools to shape soil, sand, stone, or rock. When you do, you can use your Prime Modifier instead of the normal Attribute on any Checks you make.' }]
 	},
 	{
 		id: 'terraborn_stone_blood',
@@ -2458,8 +2515,8 @@ export const traitsData: Trait[] = [
 			'You have ADV on Saves against Bleeding. Additionally, you can spend 1 AP to end the Bleeding Condition on yourself.',
 		cost: 1,
 		effects: [
-			{ type: 'GRANT_ADV_ON_SAVE', target: 'Bleeding', value: true },
-			{ type: 'GRANT_ABILITY', target: 'End Bleeding (1 AP)', value: true }
+			{ type: 'GRANT_ADV_ON_SAVE', target: 'Bleeding', value: 'ADV' },
+			{ type: 'GRANT_ABILITY', target: 'End Bleeding (1 AP)', value: 'You can spend 1 AP to end the Bleeding Condition on yourself.' }
 		]
 	},
 	{
@@ -2471,7 +2528,7 @@ export const traitsData: Trait[] = [
 			{
 				type: 'GRANT_ADV_ON_CHECK',
 				target: 'Stealth_Survival_in_Caves_Mines',
-				value: true
+				value: 'ADV'
 			}
 		]
 	},
@@ -2480,7 +2537,7 @@ export const traitsData: Trait[] = [
 		name: 'Earth Walker',
 		description: "You're not affected by Difficult Terrain created by rock, dirt, mud, or sand.",
 		cost: 1,
-		effects: [{ type: 'GRANT_ABILITY', target: 'Earth Walker', value: true }]
+		effects: [{ type: 'GRANT_ABILITY', target: 'Earth Walker', value: 'You\'re not affected by Difficult Terrain created by rock, dirt, mud, or sand.' }]
 	},
 	{
 		id: 'terraborn_superior_tremorsense',
@@ -2727,8 +2784,7 @@ export const traitsData: Trait[] = [
 			{
 				type: 'GRANT_SPELL',
 				target: 'any',
-				value: 1,
-				userChoice: { prompt: 'Choose 1 spell with the Psychic or Gravity tag.' }
+				value: 1
 			},
 			{
 				type: 'GRANT_ABILITY',
@@ -2767,7 +2823,8 @@ export const traitsData: Trait[] = [
 		cost: -1,
 		isNegative: true,
 		effects: [{ type: 'MODIFY_STAT', target: 'hpMax', value: -2 }]
-	}
+	},
+	
 ];
 
 // Helper functions for accessing trait data
